qds_sdk/commands.py

```diff
@@ -179,16 +180,21 @@ def get_results(self, fp=sys.stdout, inline=True, delim=None):
                     # Can this happen? Don't know what's the right thing to do in this case.
                     pass
         else:
-            acc = Account.find()
-            boto_conn = boto.connect_s3(aws_access_key_id=acc.storage_access_key,
-                                        aws_secret_access_key=acc.storage_secret_key)
-
-            log.info("Starting download from result locations: [%s]" % ",".join(r['result_location']))
-            #fetch latest value of num_result_dir
-            num_result_dir = Command.find(self.id).num_result_dir
-            for s3_path in r['result_location']:
-                # In Python 3, in this case, `fp` should always be binary mode.
-                _download_to_local(boto_conn, s3_path, fp, num_result_dir, delim=delim)
+            if fetch:
+                acc = Account.find()
+                boto_conn = boto.connect_s3(aws_access_key_id=acc.storage_access_key,
+                                            aws_secret_access_key=acc.storage_secret_key)
+
+                log.info("Starting download from result locations: [%s]" % ",".join(r['result_location']))
+                #fetch latest value of num_result_dir
+                num_result_dir = Command.find(self.id).num_result_dir
+                for s3_path in r['result_location']:
+                    # In Python 3, in this case, `fp` should always be binary mode.
+                    _download_to_local(boto_conn, s3_path, fp, num_result_dir, delim=delim)
+            else:
+                log.info("Returning Result Locations: [%s]" % ",".join(r['result_location']))
```

Comment:
Will remove the log and write to the file directly.

As an aside - I think the get_results() behaviour of reading s3 blobs is not desirable. let me know what you think. This potentially starts reading large result sets (think >1GB) and starts downloading them. It doesn't have retries to handle AWS API failures either. Do you think fetch=False should be made default? Is this behaviour expected right now by users of the SDK?